google/cloud/ndb/_retry.py

```diff
@@ -115,9 +115,9 @@ def is_transient_error(error):
     if core_retry.if_transient_error(error):
         return True
 
-    method = getattr(error, "code", None)
-    if method is not None:
-        code = method()
+    code_attr = getattr(error, "code", None)
+    if code_attr is not None:
+        code = code_attr() if callable(code_attr) else code_attr
```

Comment:
I think it can be done this way. It'll still give `False` for errors which are not in `TRANSIENT_CODES`, but if any new `int` error code will pretend to be added into `TRANSIENT_CODES`, it could be added easily without changing function code.